airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -574,7 +596,40 @@ def cast_property_type(self, property_name: str, column_name: str, jinja_column:
                 return f'nullif(cast({column_name} as {sql_type}), "") as {column_name}'
             replace_operation = jinja_call(f"empty_string_to_null({jinja_column})")
             return f"cast({replace_operation} as {sql_type}) as {column_name}"
-        elif is_string(definition["type"]):
+        elif (data_type.REF_TYPE_VAR_NAME in definition and is_binary_datatype(definition[data_type.REF_TYPE_VAR_NAME])) or (
+            data_type.ONE_OF_VAR_NAME in definition and is_binary_datatype(definition)
+        ):
+            if self.destination_type.value == DestinationType.POSTGRES.value:
+                # sql_type = "bytea"
+                sql_type = jinja_call("type_binary()")
+                return f"cast(decode({column_name}, 'base64') as {sql_type}) as {column_name}"
+            elif self.destination_type.value == DestinationType.BIGQUERY.value:
+                # sql_type = "bytes"
+                sql_type = jinja_call("type_binary()")
+                return f"cast(FROM_BASE64({column_name}) as {sql_type}) as {column_name}"
+            elif self.destination_type.value == DestinationType.MYSQL.value or self.destination_type.value == DestinationType.TIDB.value:
+                # sql_type = "BINARY"
+                sql_type = jinja_call("type_binary()")
+                return f"cast(FROM_BASE64({column_name}) as {sql_type}) as {column_name}"
+            elif self.destination_type.value == DestinationType.MSSQL.value:
+                # sql_type = "VARBINARY(MAX)"
+                sql_type = jinja_call("type_binary()")
+                return f"CAST({column_name} as XML ).value('.','{sql_type}') as {column_name}"
+            elif self.destination_type.value == DestinationType.SNOWFLAKE.value:
+                # sql_type = "VARBINARY"
+                sql_type = jinja_call("type_binary()")
+                return f"cast(BASE64_DECODE_BINARY({column_name}) as {sql_type}) as {column_name}"
+            elif self.destination_type.value == DestinationType.CLICKHOUSE.value:
+                # sql_type = "VARBINARY"
+                sql_type = jinja_call("type_binary()")
+                trimmed_column_name = f"trim(BOTH '\"' from {column_name})"
```

Comment:
what does this do?

Reply:
This is how clickhouse works. It complains about " symbols in value. So need to trim it explicitly.